src/Typst/Parse.hs

```diff
@@ -984,13 +984,13 @@ pArrayExpr =
       )
         <|> (Array [] <$ optional (void $ sym ","))
 
--- dict-expr ::= '(' (':' | (pair (',' pair)* ','?)) ')'
+-- dict-expr ::= '(' (':' | ':'? (pair (',' pair)* ','?)) ')'
 -- pair ::= (ident | str) ':' expr
 pDictExpr :: P Expr
-pDictExpr = try $ inParens (pEmptyDict <|> pNonemptyDict)
+pDictExpr = try $ inParens (try pNonemptyDict <|> pEmptyDict)
   where
     pEmptyDict = Dict mempty <$ sym ":"
-    pNonemptyDict = Dict <$> sepEndBy1 (pSpread <|> pPair) (sym ",")
+    pNonemptyDict = Dict <$> (optional (sym ":") *> sepEndBy1 (pSpread <|> pPair) (sym ","))
```

Comment:
I think this might be just *slightly* better (warning: untested):
```hs
pDictExpr = try $ inParens ((sym ":" *> (pNonemptyDict <|> pure (Dict mempty))) <|> pNonemptyDict)
pNonemptyDict = Dict <$> sepEndBy1 (pSpread <|> pPair) (sym ",")
```
This avoids backtracking and parsing the `:` twice.  I'll leave it up to you whether you want to change it.

Reply:
I think this is a good change; I changed my code to this